File: dmn-test-cases/standard/tck/1.3/cl3/0085-decision-services/translator/expected/java/dmn/Decision_011_2.java

import java.util.*;
import java.util.stream.Collectors;

@javax.annotation.Generated(value = {"decision.ftl", "decision_011_2"})
@com.gs.dmn.runtime.annotation.DRGElement(
    namespace = "",
    name = "decision_011_2",
    label = "",
    elementKind = com.gs.dmn.runtime.annotation.DRGElementKind.DECISION,
    expressionKind = com.gs.dmn.runtime.annotation.ExpressionKind.LITERAL_EXPRESSION,
    hitPolicy = com.gs.dmn.runtime.annotation.HitPolicy.UNKNOWN,
    rulesCount = -1
)
public class Decision_011_2 extends com.gs.dmn.runtime.JavaTimeDMNBaseDecision<String> {
    public static final com.gs.dmn.runtime.listener.DRGElement DRG_ELEMENT_METADATA = new com.gs.dmn.runtime.listener.DRGElement(
        "",
        "decision_011_2",
        "",
        com.gs.dmn.runtime.annotation.DRGElementKind.DECISION,
        com.gs.dmn.runtime.annotation.ExpressionKind.LITERAL_EXPRESSION,
        com.gs.dmn.runtime.annotation.HitPolicy.UNKNOWN,
        -1
    );

    private final Decision_011_3 decision_011_3;
    private final Decision_011_4 decision_011_4;

    public Decision_011_2() {
        this(new Decision_011_3(), new Decision_011_4());
    }

    public Decision_011_2(Decision_011_3 decision_011_3, Decision_011_4 decision_011_4) {
        this.decision_011_3 = decision_011_3;
        this.decision_011_4 = decision_011_4;
    }

    @java.lang.Override()
    public String applyMap(java.util.Map<String, String> input_, com.gs.dmn.runtime.ExecutionContext context_) {
        try {
            return apply(input_.get("inputData_011_1"), input_.get("inputData_011_2"), context_);
        } catch (Exception e) {
            logError("Cannot apply decision 'Decision_011_2'", e);
            return null;
        }
    }

    public String apply(String inputData_011_1, String inputData_011_2, com.gs.dmn.runtime.ExecutionContext context_) {
        try {
            // Start decision 'decision_011_2'
            com.gs.dmn.runtime.annotation.AnnotationSet annotationSet_ = context_ != null ? context_.getAnnotations() : null;
            com.gs.dmn.runtime.listener.EventListener eventListener_ = context_ != null ? context_.getEventListener() : null;
            com.gs.dmn.runtime.external.ExternalFunctionExecutor externalExecutor_ = context_ != null ? context_.getExternalFunctionExecutor() : null;
            com.gs.dmn.runtime.cache.Cache cache_ = context_ != null ? context_.getCache() : null;
            long decision_011_2StartTime_ = System.currentTimeMillis();
            com.gs.dmn.runtime.listener.Arguments decision_011_2Arguments_ = new com.gs.dmn.runtime.listener.Arguments();
            decision_011_2Arguments_.put("inputData_011_1", inputData_011_1);
            decision_011_2Arguments_.put("inputData_011_2", inputData_011_2);
            eventListener_.startDRGElement(DRG_ELEMENT_METADATA, decision_011_2Arguments_);

            // Evaluate decision 'decision_011_2'
            String output_ = lambda.apply(inputData_011_1, inputData_011_2, context_);

            // End decision 'decision_011_2'
            eventListener_.endDRGElement(DRG_ELEMENT_METADATA, decision_011_2Arguments_, output_, (System.currentTimeMillis() - decision_011_2StartTime_));

            return output_;
        } catch (Exception e) {
            logError("Exception caught in 'decision_011_2' evaluation", e);
            return null;
        }
    }

    public com.gs.dmn.runtime.LambdaExpression<String> lambda =
        new com.gs.dmn.runtime.LambdaExpression<String>() {
            public String apply(Object... args_) {
                String inputData_011_1 = 0 < args_.length ? (String) args_[0] : null;
                String inputData_011_2 = 1 < args_.length ? (String) args_[1] : null;
                com.gs.dmn.runtime.ExecutionContext context_ = 2 < args_.length ? (com.gs.dmn.runtime.ExecutionContext) args_[2] : null;
                com.gs.dmn.runtime.annotation.AnnotationSet annotationSet_ = context_ != null ? context_.getAnnotations() : null;
                com.gs.dmn.runtime.listener.EventListener eventListener_ = context_ != null ? context_.getEventListener() : null;
                com.gs.dmn.runtime.external.ExternalFunctionExecutor externalExecutor_ = context_ != null ? context_.getExternalFunctionExecutor() : null;
                com.gs.dmn.runtime.cache.Cache cache_ = context_ != null ? context_.getCache() : null;

                // Apply child decisions
                String decision_011_3 = Decision_011_2.this.decision_011_3.apply(context_);
                String decision_011_4 = Decision_011_2.this.decision_011_4.apply(context_);

                return stringAdd(stringAdd(stringAdd(stringAdd(stringAdd(stringAdd(inputData_011_1, " "), inputData_011_2), " "), decision_011_3), " "), decision_011_4);
            }
        };
}

File: dmn-test-cases/standard/tck/1.3/cl3/0085-decision-services/translator/expected/python/dmn/Decision_011_2.py
import typing
import decimal
import datetime
import isodate
import time

import jdmn.runtime.Context
import jdmn.runtime.DefaultDMNBaseDecision
import jdmn.runtime.ExecutionContext
import jdmn.runtime.LambdaExpression
import jdmn.runtime.LazyEval
import jdmn.runtime.Pair
import jdmn.runtime.Range
import jdmn.runtime.RuleOutput
import jdmn.runtime.RuleOutputList

import jdmn.runtime.annotation.Annotation
import jdmn.runtime.annotation.AnnotationSet
import jdmn.runtime.annotation.DRGElementKind
import jdmn.runtime.annotation.ExpressionKind
import jdmn.runtime.annotation.HitPolicy

import jdmn.runtime.cache.Cache

import jdmn.runtime.external.ExternalFunctionExecutor

import jdmn.runtime.listener.Arguments
import jdmn.runtime.listener.DRGElement
import jdmn.runtime.listener.EventListener
import jdmn.runtime.listener.Rule

import Decision_011_3
import Decision_011_4


# Generated(value = ["decision.ftl", "decision_011_2"])
class Decision_011_2(jdmn.runtime.DefaultDMNBaseDecision.DefaultDMNBaseDecision):
    DRG_ELEMENT_METADATA: jdmn.runtime.listener.DRGElement.DRGElement = jdmn.runtime.listener.DRGElement.DRGElement(
        "",
        "decision_011_2",
        "",
        jdmn.runtime.annotation.DRGElementKind.DRGElementKind.DECISION,
        jdmn.runtime.annotation.ExpressionKind.ExpressionKind.LITERAL_EXPRESSION,
        jdmn.runtime.annotation.HitPolicy.HitPolicy.UNKNOWN,
        -1
    )

    def __init__(self, decision_011_3: Decision_011_3.Decision_011_3 = None, decision_011_4: Decision_011_4.Decision_011_4 = None):
        jdmn.runtime.DefaultDMNBaseDecision.DefaultDMNBaseDecision.__init__(self)
        self.decision_011_3 = Decision_011_3.Decision_011_3() if decision_011_3 is None else decision_011_3
        self.decision_011_4 = Decision_011_4.Decision_011_4() if decision_011_4 is None else decision_011_4

    def apply(self, inputData_011_1: typing.Optional[str], inputData_011_2: typing.Optional[str], annotationSet_: jdmn.runtime.annotation.AnnotationSet.AnnotationSet, eventListener_: jdmn.runtime.listener.EventListener.EventListener, externalExecutor_: jdmn.runtime.external.ExternalFunctionExecutor.ExternalFunctionExecutor, cache_: jdmn.runtime.cache.Cache.Cache) -> typing.Optional[str]:
        try:
            # Start decision 'decision_011_2'
            decision_011_2StartTime_ = int(time.time_ns()/1000)
            decision_011_2Arguments_ = jdmn.runtime.listener.Arguments.Arguments()
            decision_011_2Arguments_.put("inputData_011_1", inputData_011_1)
            decision_011_2Arguments_.put("inputData_011_2", inputData_011_2)
            eventListener_.startDRGElement(self.DRG_ELEMENT_METADATA, decision_011_2Arguments_)

            # Evaluate decision 'decision_011_2'
            output_: typing.Optional[str] = self.evaluate(inputData_011_1, inputData_011_2, annotationSet_, eventListener_, externalExecutor_, cache_)

            # End decision 'decision_011_2'
            eventListener_.endDRGElement(self.DRG_ELEMENT_METADATA, decision_011_2Arguments_, output_, (int(time.time_ns()/1000) - decision_011_2StartTime_))

            return output_
        except Exception as e:
            self.logError("Exception caught in 'decision_011_2' evaluation", e)
            return None

    def evaluate(self, inputData_011_1: typing.Optional[str], inputData_011_2: typing.Optional[str], annotationSet_: jdmn.runtime.annotation.AnnotationSet.AnnotationSet, eventListener_: jdmn.runtime.listener.EventListener.EventListener, externalExecutor_: jdmn.runtime.external.ExternalFunctionExecutor.ExternalFunctionExecutor, cache_: jdmn.runtime.cache.Cache.Cache) -> typing.Optional[str]:
        # Apply child decisions
        decision_011_3: typing.Optional[str] = self.decision_011_3.apply(annotationSet_, eventListener_, externalExecutor_, cache_)
        decision_011_4: typing.Optional[str] = self.decision_011_4.apply(annotationSet_, eventListener_, externalExecutor_, cache_)

        return self.stringAdd(self.stringAdd(self.stringAdd(self.stringAdd(self.stringAdd(self.stringAdd(inputData_011_1, " "), inputData_011_2), " "), decision_011_3), " "), decision_011_4)
